spec/controllers/reservations_controller_spec.rb

```diff
@@ -1197,26 +1197,68 @@
             end
 
             it "allows it to start" do
-              expect(assigns(:reservation)).to_not be_changed
               expect(assigns(:reservation).actual_start_at).to eq(Time.zone.now)
             end
           end
 
-          context "and there is another reservation still going on" do
+          context "in the grace period, but there is another reservation still running" do
             let!(:reservation2) do
               create(:purchased_reservation, product: @instrument,
-                                             reserve_start_at: start_at - 30.minutes, reserve_end_at: start_at)
+                                             reserve_start_at: start_at - 30.minutes,
+                                             reserve_end_at: start_at,
+                                             actual_start_at: start_at - 30.minutes)
+            end
+
+            it "does not start the reservation" do
+              do_request
+              expect(assigns(:reservation)).not_to be_started
+              expect(flash[:error]).to match(/previously scheduled reservation is ongoing/)
+            end
+          end
+
+          context "and there is a non-started reservation" do
+            let!(:reservation2) do
+              create(:purchased_reservation, product: @instrument,
+                                             reserve_start_at: start_at - 30.minutes,
+                                             reserve_end_at: start_at,
+                                             actual_start_at: nil)
             end
 
             it "allows it to start" do
               do_request
-              expect(assigns(:reservation)).to_not be_changed
-              expect(assigns(:reservation).reserve_start_at).to eq(start_at)
               expect(assigns(:reservation).actual_start_at).to eq(Time.zone.now)
             end
           end
         end
 
+        context "when there is a prior reservation overrunning its time" do
+          let(:start_at) { 3.minutes.ago.change(usec: 0) }
+          let(:end_at) { 57.minutes.from_now.change(usec: 0) }
```

Comment:
Just curious: why are the `change(usec: 0)` calls here needed?

Reply:
It's a copy and paste from the a spec or two down. I'm not 100% sure, but I'm pretty sure it has to do with database vs ruby timestamp precision--I'm pretty sure mysql truncates the milliseconds, or at least it did at one point. I can probably get away without it here since I'm not directly comparing timestamps in any of my new tests.